### PR TITLE
Add cluster endpoint on the search API server.

### PR DESCRIPTION
**Pull request by mosuka**
### Context / purpose
Add cluster endpoint on the search API server. #354 

### Description
Add an endpoint to expose the cluster membership information.

### How was this PR tested?
```
cargo run -- serve --host=0.0.0.0 --index-uri=s3://quickwit-dev/hdfslogs-idx --port=8080  
cargo run -- serve --host=0.0.0.0 --index-uri=s3://quickwit-dev/hdfslogs-idx --port=8090 --peer-seed 127.0.0.1:8080  
cargo run -- serve --host=0.0.0.0 --index-uri=s3://quickwit-dev/hdfslogs-idx --port=8100 --peer-seed 127.0.0.1:8080 
```

```
curl 'http://0.0.0.0:8080/cluster/members?format=prettyjson'
```

```
{
  "members": [
    {
      "id": "9804b14f-2cb2-4cc7-aa91-5e85d2ce27d9",
      "listenAddress": "0.0.0.0:8081",
      "isSelf": true
    },
    {
      "id": "c8d1aad5-aeca-42f3-9e1b-9b1de9a89f53",
      "listenAddress": "127.0.0.1:8091",
      "isSelf": false
    },
    {
      "id": "d9b68439-4c6a-40f7-aea0-19f3238e293f",
      "listenAddress": "127.0.0.1:8101",
      "isSelf": false
    }
  ]
}
```

### Checklist
*Remove or complete this list if required.*
- [x] tested locally

### CODE REVIEW

File: quickwit-serve/src/grpc_adapter/cluster_adapter.rs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+use std::sync::Arc;
+
+use async_trait::async_trait;
+
+use quickwit_cluster::error::ClusterError;
+use quickwit_cluster::service::{ClusterService, ClusterServiceImpl};
+use quickwit_proto::cluster_service_server as grpc;
+
+/// gRPC adapter that wraped SearchService.
```
Comment (fulmicoton) [inline quickwit-serve/src/grpc_adapter/cluster_adapter.rs:30]: remove comment, it is not helping.

File: quickwit-serve/src/grpc_adapter/search_adapter.rs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+use async_trait::async_trait;
+use tokio_stream::wrappers::UnboundedReceiverStream;
+
+use quickwit_proto::{
+    search_service_server as grpc, LeafSearchStreamRequest, LeafSearchStreamResult,
+};
+use quickwit_search::{SearchError, SearchService, SearchServiceImpl};
+
+/// gRPC adapter that wraped SearchService.
```
Comment (fulmicoton) [inline quickwit-serve/src/grpc_adapter/search_adapter.rs:32]: remove comment, it is not helping.

File: quickwit-serve/src/http_handler/cluster.rs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+async fn members_endpoint<TClusterService: ClusterService>(
+    cluster_service: &TClusterService,
+) -> Result<quickwit_proto::MembersResult, ApiError> {
+    let members_request = quickwit_proto::MembersRequest {};
+    let members_result = cluster_service.members(members_request).await?;
+    Ok(members_result)
+}
+
+fn make_reply<T: serde::Serialize>(result: Result<T, ApiError>) -> impl Reply {
```
Comment (fulmicoton) [inline quickwit-serve/src/http_handler/cluster.rs:63]: I think this is copy pasted from somewhere else. Can we factorize this?

**Review by fulmicoton — APPROVED** [review]

File: quickwit-serve/src/http_handler/cluster.rs (hunk 3 above)
Comment (mosuka) [reply quickwit-serve/src/http_handler/cluster.rs]: @fulmicoton 
Changed the members api to use the same make_reply() as the search API.
Can you please take another look?

**Pull request merged**